platforms/ide/problems-api/src/main/java/org/gradle/api/problems/internal/DefaultProblems.java

```diff
@@ -73,4 +77,24 @@ public InternalProblemReporter getInternalReporter() {
     public Multimap<Throwable, Problem> getProblemsForThrowables() {
         return problemsForThrowables;
     }
+
+    @Override
+    public void reportMapping(Throwable throwable) {
+        ImmutableMap.Builder<Throwable, Collection<Problem>> builder = ImmutableMap.<Throwable, Collection<Problem>>builder();
+
+        Throwable cause = throwable;
+
+        // TODO: handle MultiCauseException
+        // TODO: handle cyclic causes (ie. this == cause and this.cause.cause == this)
+
+        while (cause != null) {
+            for (Map.Entry<Throwable, Collection<Problem>> entry : problemsForThrowables.asMap().entrySet()) {
+                if (entry.getKey().equals(cause)) {
```

Comment:
🤔 Why not call `problemsForThrowables.get(cause)`?

Reply:
good question :)
I'll fix that.